### PR TITLE
add boundary scanner and max fragment length support to unified highl…

### PR DESCRIPTION
**Pull request by Mpdreamz**
…igther as per elastic/elasticsearch#23431

### CODE REVIEW

File: src/Nest/Search/Search/Highlighting/Highlight.cs [hunk 1]
```diff
@@ -115,5 +148,13 @@ public class HighlightDescriptor<T> : DescriptorBase<HighlightDescriptor<T> ,IHi
 		public HighlightDescriptor<T> BoundaryCharacters(string boundaryCharacters) => Assign(a => a.BoundaryChars = boundaryCharacters);
 
 		public HighlightDescriptor<T> BoundaryMaxSize(int boundaryMaxSize) => Assign(a => a.BoundaryMaxSize = boundaryMaxSize);
```
Comment (russcam) [inline src/Nest/Search/Search/Highlighting/Highlight.cs]: I realise this is not part of the changes here, but it looks like `boundary_max_size` should be `boundary_max_scan`:

https://github.com/elastic/elasticsearch/blob/4863a9cfdf2ca5a2fdb503e5dab9372a0462cd20/core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/AbstractHighlighterBuilder.java#L64-L64

I think it makes sense to update as part of this PR

Comment (Mpdreamz) [inline src/Nest/Search/Search/Highlighting/Highlight.cs]: https://github.com/elastic/elasticsearch/issues/1614 we're in good company of making this mistake :)

Comment (Mpdreamz) [inline src/Nest/Search/Search/Highlighting/Highlight.cs]: added _scan and obsoleted _size

**Comment by Mpdreamz** [comment]
Good catch @russcam added support for `fragmenter` as well.

**Comment by russcam** [comment]
++ I think `fragmenter` should also be added to the highlighter tests

**Comment by Mpdreamz** [comment]
@russcam added test and also copied the properties to `IHighlightField` as well.

**Comment by Mpdreamz** [comment]
ported to `master`

**Pull request merged**